packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
- *  isSelected={true}
- *  isNew={false}
- * />
- */
-export const Tile = Object.assign(
-  React.forwardRef((props, forwardedRef) => {
-    const {
-      className,
-      name,
-      description,
-      metadata,
-      thumbnail,
-      buttons,
-      leftIcon,
-      rightIcon,
-      badge,
-      isNew,
-      isSelected,
-      moreOptions,
-      variant = 'default',
-      children,
-      isActionable: isActionableProp,
-      status,
-      isLoading = false,
-      isDisabled = false,
-      onClick,
-      ...rest
-    } = props;
-
-    const [isMenuVisible, setIsMenuVisible] = React.useState(false);
-    const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
-    const hideMenu = React.useCallback(() => setIsMenuVisible(false), []);
-    const [localActionable, setLocalActionable] =
-      React.useState(isActionableProp);
-
-    const isActionable = isActionableProp ?? localActionable;
-
-    const tileName = (
-      <Box className='iui-tile-name'>
-        <TitleIcon
-          isLoading={isLoading}
-          isSelected={isSelected}
-          isNew={isNew}
-          status={status}
-        />
-
-        <Box as='span' className='iui-tile-name-label'>
-          {isActionable && onClick ? (
-            <LinkAction
-              as='button'
-              onClick={!isDisabled ? onClick : undefined}
-              aria-disabled={isDisabled}
-            >
-              {name}
-            </LinkAction>
-          ) : (
-            name
-          )}
-        </Box>
-      </Box>
-    );
+const TileComponent = React.forwardRef((props, ref) => {
+  const {
+    className,
+    status,
+    variant,
+    isNew,
+    isSelected,
+    isLoading,
+    isActionable: isActionableProp,
+    isDisabled,
+    onClick,
+    ...rest
+  } = props;
+  const [localActionable, setLocalActionable] =
+    React.useState(isActionableProp);
+  const isActionable = isActionableProp ?? localActionable;
+
+  return (
+    <TileContext.Provider
+      value={{
+        status,
+        variant,
+        isNew,
+        isSelected,
+        isLoading,
+        isActionable,
+        isDisabled,
+        setActionable: setLocalActionable,
+        onClick,
+      }}
+    >
+      <Box
+        className={cx(
+          'iui-tile',
+          {
+            'iui-folder': variant === 'folder',
+            'iui-new': isNew,
+            'iui-selected': isSelected,
+            'iui-actionable': isActionable,
+            [`iui-${status}`]: !!status,
+            'iui-loading': isLoading,
+          },
+          className,
+        )}
+        aria-disabled={isDisabled}
+        ref={ref}
+        {...rest}
+      />
+    </TileContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileOwnProps>;
+TileComponent.displayName = 'Tile';
+
+// ----------------------------------------------------------------------------
+// Tile.Action component
+
+const TileAction = React.forwardRef((props, forwardedRef) => {
```

Comment:
nitpicky, but `forwardedRef` can change to `ref` for consistency

Reply:
forwardedRef is the new preferred name

https://github.com/iTwin/iTwinUI/blob/ab07a844f5c2bca1c255adc5502e572c34fc8ede/scripts/createComponent.js#L71

Reply:
okay, so I updated to `forwardedRef` for all of them